src/sagemaker/clarify.py

```diff
@@ -393,6 +431,94 @@ def to_dict(self) -> Dict[str, Any]:  # pragma: no cover
+                series in the shared input dataset.
+            item_metadata (list[str] or list[int]): Optional.  An array of strings or
+                array of zero-based integer indices. Used to locate all item metadata
+                fields in the shared input dataset.
+
+        Raises:
+            AssertionError: If any required arguments are not provided.
+            ValueError: If any provided arguments are the wrong type.
+        """
+        # check target_time_series, item_id, and timestamp are provided
+        assert target_time_series, "Please provide a target time series."
+        assert item_id, "Please provide an item id."
+        assert timestamp, "Please provide a timestamp."
+        # check all arguments are the right types
+        if not isinstance(target_time_series, (str, int)):
+            raise ValueError("Please provide a string or an int for ``target_time_series``")
+        if not isinstance(item_id, (str, int)):
+            raise ValueError("Please provide a string or an int for ``item_id``")
+        if not isinstance(timestamp, (str, int)):
+            raise ValueError("Please provide a string or an int for ``timestamp``")
+        # add remaining fields to an internal dictionary
+        self.analysis_config = dict()
+        _set(target_time_series, "target_time_series", self.analysis_config)
+        _set(item_id, "item_id", self.analysis_config)
+        _set(timestamp, "timestamp", self.analysis_config)
+        # check optional arguments are right types if provided
+        related_time_series_error_message = (
+            "Please provide a list of strings or list of ints for ``related_time_series``"
+        )
+        if related_time_series:
+            if not isinstance(related_time_series, list):
+                raise ValueError(
+                    related_time_series_error_message
+                )  # related_time_series is not a list
+            if not (
+                all([isinstance(value, str) for value in related_time_series])
+                or all([isinstance(value, int) for value in related_time_series])
+            ):
+                raise ValueError(
+                    related_time_series_error_message
+                )  # related_time_series is not a list of strings or list of ints
+            _set(
+                related_time_series, "related_time_series", self.analysis_config
+            )  # related_time_series is valid, add it
+        item_metadata_series_error_message = (
+            "Please provide a list of strings or list of ints for ``item_metadata``"
+        )
+        if item_metadata:
+            if not isinstance(item_metadata, list):
+                raise ValueError(item_metadata_series_error_message)  # item_metadata is not a list
+            if not (
+                all([isinstance(value, str) for value in item_metadata])
+                or all([isinstance(value, int) for value in item_metadata])
```

Comment:
To be accurate, it is even more. 

It is either ALL attributes (tts, rts, imd, etc.) should be `strings` OR ALL `ints`